cpp/tests/binaryop/util/operation.h

```diff
@@ -143,9 +143,48 @@ struct TrueDiv {
 
 template <typename TypeOut, typename TypeLhs, typename TypeRhs>
 struct FloorDiv {
+  template <typename OutT                                                         = TypeOut,
+            typename LhsT                                                         = TypeLhs,
+            typename RhsT                                                         = TypeRhs,
+            std::enable_if_t<(std::is_integral_v<std::common_type_t<LhsT, RhsT>> and
+                              std::is_signed_v<std::common_type_t<LhsT, RhsT>>)>* = nullptr>
   TypeOut operator()(TypeLhs lhs, TypeRhs rhs)
   {
-    return static_cast<TypeOut>(floor(static_cast<double>(lhs) / static_cast<double>(rhs)));
+    if ((lhs ^ rhs) >= 0) {
+      return lhs / rhs;
+    } else {
+      auto const quotient  = lhs / rhs;
+      auto const remainder = lhs % rhs;
+      return quotient - !!remainder;
+    }
+  }
+
+  template <typename OutT                                                          = TypeOut,
+            typename LhsT                                                          = TypeLhs,
+            typename RhsT                                                          = TypeRhs,
+            std::enable_if_t<(std::is_integral_v<std::common_type_t<LhsT, RhsT>> and
+                              !std::is_signed_v<std::common_type_t<LhsT, RhsT>>)>* = nullptr>
+  TypeOut operator()(TypeLhs lhs, TypeRhs rhs)
+  {
+    return lhs / rhs;
+  }
+
+  template <typename OutT                                                              = TypeOut,
+            typename LhsT                                                              = TypeLhs,
+            typename RhsT                                                              = TypeRhs,
+            std::enable_if_t<(std::is_same_v<std::common_type_t<LhsT, RhsT>, float>)>* = nullptr>
+  TypeOut operator()(TypeLhs lhs, TypeRhs rhs)
+  {
+    return static_cast<TypeOut>(std::floor(lhs / rhs));
```

Comment:
```suggestion
    return static_cast<TypeOut>(std::floorf(lhs / rhs));
```

Reply:
Per https://en.cppreference.com/w/cpp/numeric/math/floor I think there's an overload for `std::floor` with a float argument?

Reply:
Yeah, my intention is to use a more descriptive name so people know it's dedicated to `float` operations. Not necessarily a change request.